Add soft keywords to Python keyword list

eht16 opened the pull request:
This adds `_`, `case` and `match` to the Python keywords list.

`_` seems not very useful but also won't hurt I guess. It will be highlighted as keyword when used in something like `message = _('Text to be translated')`, while not completely technically correct in this case, it is OK IMO and not worth the efforts to remove this single keyword from the list in the update script.

N.B.: the `scripts/update-python-identifiers.sh` script now requires Python 3.10 which was released about one year ago and the script is only to be executed by developers.

Closes #3263.

Before: 
![issue3263_before](https://user-images.githubusercontent.com/617017/188320292-8565106a-bded-461d-b4cd-a8a41b0317c3.png)

After: 
![issue3263_after](https://user-images.githubusercontent.com/617017/188320298-f85d2dc4-0721-4385-9330-4727845a5aa3.png)

elextr commented:
Looks ok to me, the Python experts seem to be in agreement that this should be done so ok.

Committ message should also mention "... and update to Python 3.10" since it also removed `__peg_parser__` and added to identifiers as well as adding `match` and `case`.

eht16 commented:
> Looks ok to me, the Python experts seem to be in agreement that this should be done so ok.
> 
> Committ message should also mention "... and update to Python 3.10" since it also removed `__peg_parser__` and added to identifiers as well as adding `match` and `case`.

Thanks, done.

eht16 commented:
If there won't be objections, I'll merge this in a few days.
